### PR TITLE
feat(models): add User and Note type references

### PR DESCRIPTION
**Pull request by joematune**
**tldr**: This PR includes semantic TS only changes

- Add type references between models - 470c91be42847daeba7eadd17806f560da841faf

Reasoning: As Dynamo models grow, the ability to add relations between models can be illustrated and documented with TS. This is especially true for single table design patterns as seen below where we can see that a note record's `pk` is a reference to the user's `id`, which is of type: `note#${string}`. The note's `id` on the other hand is a `cuid` which is documented with type `ReturnType<typeof cuid>`

Screen Shot 2022-03-18 at 11 06 17
Screen Shot 2022-03-18 at 11 06 45

### CODE REVIEW

File: app/models/note.server.ts [hunk 1]
```diff
@@ -1,22 +1,28 @@
-import cuid from "cuid";
 import arc from "@architect/functions";
+import cuid from "cuid";
+import { User } from "./user.server";
```
Comment (MichaelDeBoey) [inline app/models/note.server.ts:3]: Since we're only using the type, let's import just the type

```suggestion

import type { User } from "./user.server";
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

**Review by MichaelDeBoey — APPROVED** [review]

**Review by kentcdodds — APPROVED** [review]
I'm a fan 👍

**Comment by kentcdodds** [comment]
Could you take a look at the type errors in this PR?

**Comment by joematune** [comment]
> Could you take a look at the type errors in this PR?

@kentcdodds Will squash them 👍

**Comment by joematune** [comment]
@kentcdodds - Tidied the errors here - [aa57c40](https://github.com/remix-run/grunge-stack/pull/22/commits/aa57c401bb7dd57a34dcd5d2313b30edcd3c82b0)

I ran `npm run typecheck`, which was the point of failure in the [previous run of actions](https://github.com/remix-run/grunge-stack/runs/5604869869?check_suite_focus=true), and it came out clean. Should be solid this time 🙆

**Review by kentcdodds — APPROVED** [review]
Thank you 👏
